src/auth/KeyRing.cc

```diff
@@ -203,6 +203,12 @@ void KeyRing::decode(bufferlist::const_iterator& bl) {
   } catch (buffer::error& err) {
     keys.clear();
     decode_plaintext(start_pos);
+  }catch (buffer::end_of_buffer& err) {
```

Comment:
nit: add space between } and catch?

Reply:
i don't understand why shall we catch `buffer::end_of_buffer` if we already catch `buffer::error`. as the former is derived from the latter.

Reply:
reading the description of the tracker ticket

> Getting out of bounds access in the buffer, generating a trap.

accessing out of array bounds is undefined behavior. how can we catch it? what exactly exception is thrown?